snapcraft/internal/repo/_deb.py

```diff
@@ -517,6 +521,27 @@ def _mark_package_dependencies(
                 target_arch=target_arch,
             )
 
+    @classmethod
+    @functools.lru_cache(maxsize=256)
+    def get_packages_in_base(cls, *, base: str) -> List[str]:
+        # We do not want to break what we already have.
+        if base in ("core", "core16", "core18"):
+            return _DEFAULT_FILTERED_STAGE_PACKAGES
+
+        base_package_list_path = _get_dpkg_list_path(base)
+        if not base_package_list_path.exists():
+            return list()
+
+        # format of package list is <package_name>[:<architecture>]
```

Comment:
`ii  adduser                       3.118ubuntu1               all          add and rem`